client/state/data-layer/utils.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import {
+	isArray,
+	mergeWith,
+} from 'lodash';
+
+const concatHandlers = ( left, right ) =>
+	isArray( left )
+		? left.concat( right )
+		: undefined;
+
+export const mergeHandlers = ( ...handlers ) =>
+	handlers.length > 1
+		? mergeWith( {}, ...handlers, concatHandlers )
+		: handlers.shift();
```

Comment:
Would `handlers[ 0 ]` (`head( handlers )`) be clearer? Took me a moment to figure out what this was doing.